hostsfile/src/lib.rs

```diff
@@ -171,6 +171,24 @@ impl HostsBuilder {
         Ok(hosts_dir.with_file_name(temp_filename))
     }
 
+    fn has_content_changed(hosts_path: &Path, new_contents: &[u8]) -> io::Result<bool> {
+        let hosts_file = File::open(hosts_path)?;
+
+        if hosts_file.metadata()?.len() != new_contents.len() as u64 {
+            return Ok(true);
+        }
+
+        let hosts_buf = BufReader::new(hosts_file);
```

Comment:
Indeed the file doesn't have to be read again! I just implemented your suggestion in a new commit, see what you think of it :)

Reply:
Yeah looking better, though I've added a comment that I think that using `lines_to_insert: Vec<String>` (per above) instead of current `lines_to_insert: Vec<u8>` could simplify things even more.